### PR TITLE
Add Wallet Connector to Base Developer Docs

### PR DESCRIPTION
**Pull request by taycaldwell**
**What changed? Why?**
- Add wallet connector (RainbowKit) to docs navbar

**Screenshots**
Screenshot 2024-04-12 at 1 32 33 PM
Screenshot 2024-04-12 at 1 32 50 PM
Screenshot 2024-04-12 at 1 32 39 PM
Screenshot 2024-04-12 at 1 32 55 PM
Screenshot 2024-04-12 at 1 33 03 PM

Screenshot 2024-04-12 at 1 33 13 PM
Screenshot 2024-04-12 at 1 35 29 PM
Screenshot 2024-04-12 at 1 33 18 PM
Screenshot 2024-04-12 at 1 33 30 PM
Screenshot 2024-04-12 at 1 33 24 PM

**Notes to reviewers**
N/A

**How has it been tested?**
Manually

**Does this PR add a new token to the bridge?**

Are you adding an entry to [`assets.ts`](../apps/bridge/assets.ts)?

- [x] No, this PR does not add a new token to the bridge
- [ ] Yes, and I've confirmed this token doesn't use a bridge override
- [ ] Yes, and I've confirmed this token is an OptimismMintableERC20

If you are adding a token to the bridge, please include evidence of both confirmations above for your reviewers.

### CODE REVIEW

**Review by briandoyle81CB — CHANGES_REQUESTED** [review]
Looks great but is breaking Cafe:  http://localhost:3000/base-camp/docs/storage/storage-exercise

It's breaking because it's using wagmi 1.x and needs to be updated.

I also worry about having Base mainnet as a network.  It's going to create the risk of someone spending a lot of real ETH to do one of the exercises.  

If we need mainnet on the site, we should make sure there are safeguards.

**Comment by cb-heimdall** [comment]
Review Error for briandoyle81CB @ 2024-04-12 21:54:22 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: apps/base-docs/src/theme/NavbarItem/styles.module.css [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+.identityContainer {
+  border-radius: 3px;
+  padding: 6px 12px !important;
+  cursor: pointer;
+}
+
+[data-theme='light'] .identityContainer {
+  background-color: rgb(235, 237, 240) !important;
```
Comment (zencephalon) [inline apps/base-docs/src/theme/NavbarItem/styles.module.css:89]: Should we name these colors somehow?

File: apps/base-docs/src/theme/NavbarItem/ComponentTypes.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+                      {account && (
+                        <div className={styles.avatar}>
+                          <WalletAvatar
+                            address={account.address}
+                            ensImage={account.ensAvatar}
+                            size={24}
+                          />
+                        </div>
+                      )}
+                      {account.ensName ? account.ensName : account.displayName}
```
Comment (zencephalon) [inline apps/base-docs/src/theme/NavbarItem/ComponentTypes.js:96]: Maybe just `account.ensName ?? account.displayName`

File: apps/base-docs/src/theme/Navbar/MobileSidebar/Header/index.js [hunk 3]
```diff
@@ -15,17 +18,23 @@ function CloseButton() {
         description: 'The ARIA label for close button of mobile sidebar',
       })}
       className="clean-btn navbar-sidebar__close"
-      onClick={() => mobileSidebar.toggle()}>
+      onClick={() => mobileSidebar.toggle()}
```
Comment (zencephalon) [inline apps/base-docs/src/theme/Navbar/MobileSidebar/Header/index.js:21]: Nit, `useCallback`

**Review by wbnns — DISMISSED** [review]
When canceling the request to connect my wallet, I get an uncaught exception:

2024-04-16 at 20 55 37@2x

Tested w/ Coinbase Wallet, MetaMask and Brave on desktop.

@taycaldwell Will mobile users who have a mobile wallet installed be able to connect?

**Comment by taycaldwell** [comment]
Need to gracefully handle these wallet errors across the site. Looking into how to do that with wagmi v2 / viem, because it was happening for Base Camp stuff too.

**Comment by wbnns** [comment]
@taycaldwell

Thanks for the updates! Looks like there is a merge conflict in `apps/base-docs/tutorials/data.json`

**Review by briandoyle81CB — APPROVED** [review]

**Pull request merged**